spring-data-neo4j/src/main/java/org/springframework/data/neo4j/util/EntityUtils.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c)  [2011-2016] "Pivotal Software, Inc." / "Neo Technology" / "Graph Aware Ltd."
+ *
+ * This product is licensed to you under the Apache License, Version 2.0 (the "License").
+ * You may not use this product except in compliance with the License.
+ *
+ * This product may include a number of subcomponents with
+ * separate copyright notices and license terms. Your use of the source
+ * code for these subcomponents is subject to the terms and
+ * conditions of the subcomponent's license, as noted in the LICENSE file.
+ *
+ */
+package org.springframework.data.neo4j.util;
+
+import java.lang.reflect.Field;
+import java.util.List;
+
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.neo4j.ogm.annotation.GraphId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.Assert;
+
+/**
+ * 
+ * @author Eric Spiegelberg
+ */
+public class EntityUtils {
+
+	private static final Logger logger = LoggerFactory.getLogger(EntityUtils.class);
+	
+	/**
+	 * Determine if the domain entity is new, with "new" being defined as if the entity's GraphId property 
+	 * having a non-null value or not.
+	 * 
+	 * @param entity The domain entity.
+	 * @return Returns boolean true is the entity's GraphId property has a null value, boolean false if 
+	 * the GraphId property has a non-null value, and Boolean null if an access/argument exception 
+	 * occurred and/or if a qualifying GraphId property could not be found.
+	 */
```

Comment:
The null value of the Boolean is a concern for me. If there's an exception or a missing GraphId, imho the exception should be thrown